tests/test_non_host.py

```diff
@@ -69,3 +72,37 @@ def test_custom_interpreter_with_local_only(
     if sys.version_info >= (3, 12):
         expected -= {"setuptools", "wheel"}  # pragma: no cover
     assert found == expected, out
+
+
+def test_custom_interpreter_ensure_pythonpath_envar_is_honored(
+    tmp_path: Path,
+    mocker: MockerFixture,
+    monkeypatch: pytest.MonkeyPatch,
+    capfd: pytest.CaptureFixture[str],
+) -> None:
+    # ensures that we honor $PYTHONPATH when passing it to the custom interpreter process
+    venv_path = str(tmp_path / "venv")
+    result = virtualenv.cli_run([venv_path, "--activators", ""])
+
+    another_path = tmp_path / "another-path"
+    fake_dist = another_path / "foo-1.2.3.dist-info"
+    fake_dist.mkdir(parents=True)
+    fake_metadata = fake_dist / "METADATA"
+    with fake_metadata.open("w") as f:
+        f.write("Metadata-Version: 2.3\n" "Name: foo\n" "Version: 1.2.3\n")
+    cmd = ["", f"--python={result.creator.exe}"]
+    mocker.patch("pipdeptree._discovery.sys.argv", cmd)
+    monkeypatch.setenv("PYTHONPATH", str(another_path))
+    main()
+    out, _ = capfd.readouterr()
+    found = {i.split("==")[0] for i in out.splitlines()}
+    implementation = python_implementation()
```

Comment:
tests/test_non_host.py (92.3%): Missing lines 102,105


You need to put the right no cover pragmas per https://github.com/asottile/covdefaults?tab=readme-ov-file#platform-specific--pragma-no-cover

Reply:
Made the necessary changes; in a future PR we should look for any usages of `# pragma no cover` and see if they can also be given a tag